libflux/fluxdoc/SPEC.md

```diff
@@ -0,0 +1,284 @@
+# Writing Flux Doc Comments
+
+Flux will treat any uninterrupted series of line comments immediately preceding a function definition, package clause, builtin statement, or any declaration of an option or constant, as a doc comment.
+
+A flux doc comment consists of multiple sections, which are defined in this document. Each section should be separated from other sections by a single blank line comment. Unless otherwise noted, each section is required, and the sections must be arranged in the order in which they are presented in this document.
+
+## Documenting functions
+
+### Name and Description
+
+Doc comments for functions should begin with the function name, wrapped in backticks, followed by a brief description of what the function does or how it is used. The name and description should read as one or more full sentences.
+
+_Example:_
+
+```
+// `join` merges two input streams into a single output stream based on columns
```

Comment:
Is the reason for the backticks just for markdown? It seems to me like we could auto-backtick the first word if it matches with the function name and this might read better. It's a minor nitpick so I'm fine with the backtick, but I'm not the biggest fan of adding too much formatting to doc comments.

Reply:
I suppose we don't really need the backticks, but it's something that came out of the working group. I think we liked it because it makes it clear that it needs to match the function name exactly, i.e. `aggregateWindow` instead of `AggregateWindow`, which we saw a lot of when looking through comments in stdlib. I suppose we could just spell that out in the spec, though.